src/e2e/specs/dashboard.spec.ts

```diff
@@ -127,6 +133,94 @@ test.describe(`${process.env.E2E_TEST_ENV} - Breaches Dashboard - Headers`, () =
     await dashboardPage.popupCloseButton.click();
     await expect(dashboardPage.aboutFixedExposuresPopup).toBeHidden();
   });
+
+  test("Verify that the Apps and Services header options work correctly.", async ({
+    dashboardPage,
+    page,
+  }) => {
+    // link to testrail
+    test.info().annotations.push({
+      type: "testrail",
+      description:
+        "https://testrail.stage.mozaws.net/index.php?/cases/view/2463569",
+    });
+
+    await dashboardPage.fixedTab.click();
+    expect(page.url()).toMatch(/.*dashboard\/fixed\/?/);
+    await dashboardPage.actionNeededTab.click();
+    expect(page.url()).toMatch(/.*dashboard\/action-needed\/?/);
+
+    //apps and services button check
+    const clickOnLinkAndGoBack = async (
+      aTag: Locator,
+      host: string | RegExp = /.*/,
+      path: string | RegExp = /.*/,
+    ) => {
+      await expect(dashboardPage.appsAndServices).toBeVisible();
+      await dashboardPage.appsAndServices.click();
+      await expect(dashboardPage.appsAndServicesMenu).toBeVisible();
+      await clickOnATagCheckDomain(aTag, host, path, page);
+    };
+
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesVpn,
+      "www.mozilla.org",
+      /.*\/products\/vpn\/?.*/,
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesRelay,
+      "relay.firefox.com",
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesPocket,
+      /getpocket\.com|apps\.apple\.com|app\.adjust\.com/,
+      /.*(\/pocket-and-firefox\/?).*|.*about.*|.*pocket-stay-informed.*/,
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesFirefoxDesktop,
+      "www.mozilla.org",
+      /.*\/firefox\/new\/?.*/,
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesFirefoxMobile,
+      "www.mozilla.org",
+      /.*\/browsers\/mobile\/?.*/,
+    );
+    await clickOnLinkAndGoBack(
+      dashboardPage.servicesMozilla,
+      "www.mozilla.org",
+    );
+
+    const openProfileMenuItem = async (
+      what: Locator,
+      whatUrl: string | RegExp,
+    ) => {
+      await dashboardPage.open();
+      await dashboardPage.profileButton.click();
+      await expect(what).toBeVisible();
+      if (await what.evaluate((e) => e.hasAttribute("href"))) {
+        const href = await what.getAttribute("href");
+        expect(href).not.toBeNull();
+        await page.goto(href!);
+      } else {
+        await what.click();
+      }
+      await page.waitForURL(whatUrl);
+    };
+
+    await openProfileMenuItem(
+      dashboardPage.manageProfile,
+      /.*accounts.*settings.*/,
+    );
+    await openProfileMenuItem(
+      dashboardPage.profileSettings,
+      /.*\/user\/settings.*/,
+    );
+
+    const base_url = process.env["E2E_TEST_BASE_URL"];
+    expect(base_url).toBeTruthy();
```

Comment:
What is this testing?
Don't we fail sooner/elsewhere if the `E2E_TEST_BASE_URL` env var isn't set?

Reply:
It says that process.env[..] is an option string, so i am just checking that its non-null. You're right that it would've probably failed before that if it was indeed null. I think its presence doesnt make a difference tbh.